src/transformState.js

```diff
@@ -5,7 +5,28 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const property of action.keysToRemove) {
+          if (state.hasOwnProperty(property)) {
+            delete state[property];
+          }
+        }
```

Comment:
I think here is too much nesting, but I'm not might be wrong. Checklist, point 5.